api/README.md

```diff
@@ -0,0 +1,25 @@
+# CloudEvents API
+
+For Maven based projects, use the following dependency:
+
+```xml
+<dependency>
+    <groupId>io.cloudevents</groupId>
+    <artifactId>cloudevents-api</artifactId>
+    <version>2.0.0-SNAPSHOT</version>
+</dependency>
+```
+
+## Features
+
+This package provides the base interfaces used by the SDK. In particular:
+
+- `CloudEvent` is the main interface representing a read only CloudEvent in-memory representation
+- `Extension` represents a _materialized_ in memory representation of a CloudEvent extension
+- `SpecVersion` is an enum of CloudEvent' specification versions supported by this SDK version.
+- `CloudEventVisitor`/`CloudEventVisitable` are the interfaces used by the SDK to implement protocol bindings/event formats
+   for `CloudEvent` implementations. These interfaces are optional and, if your `CloudEvent` doesn't implement it,
```

Comment:
> if your `CloudEvent` doesn't implement it, a default implementation is provided by the SDK.

But `CloudEvent` does not _need_ to implement them. These are optional interfaces _next_ to the `CloudEvent` type (and only make me question the need for them in `api` even more).

Reply:
> But CloudEvent does not need to implement them

And it's stated pretty clearly: _These interfaces are optional_ aka if you want, implement them, otherwise the sdk will take care of it.
How do you propose to rephrase it?

Reply:
I'd clearly state the purpose of them, why they are part of `api`, who may need/want to implement them and how/why to use them (without going into `core`'s implementation details)

Reply:
Now?

Reply:
Looks good 👍